OrbitLinuxTracing/UprobesUnwindingVisitor.cpp

```diff
@@ -140,19 +182,35 @@ void UprobesUnwindingVisitor::visit(UprobePerfEventWithStack* event) {
   }
 }
 
-void UprobesUnwindingVisitor::visit(UretprobePerfEventWithStack* event) {
-  FunctionEnd function_end{event->TID(), event->GetFunction()->VirtualAddress(),
-                           event->Timestamp()};
-  if (listener_ != nullptr) {
-    listener_->OnFunctionEnd(function_end);
+void UprobesUnwindingVisitor::visit(UretprobesPerfEvent* event) {
+  std::optional<FunctionCall> function_call =
+      function_call_manager_.ProcessUretprobes(event->TID(),
+                                               event->Timestamp());
+  if (function_call.has_value()) {
```

Comment:
Do both conditions in the same if: 

if (function_call.has_value() && listener_ != nullptr) {
 ...
}

Reply:
Done.